lti_consumer/fields.py

```diff
@@ -0,0 +1,132 @@
+import logging
+
+from django.core.exceptions import ValidationError
+from django.db import models
+
+from opaque_keys.edx.keys import CourseKey
+
+log = logging.getLogger(__name__)
+
+
+def _strip_object(key):
+    """
+    Strips branch and version info if the given key supports those attributes.
+    """
+    if hasattr(key, 'version_agnostic') and hasattr(key, 'for_branch'):
+        return key.for_branch(None).version_agnostic()
+    else:
+        return key
+
+
+def _strip_value(value, lookup='exact'):
+    """
+    Helper function to remove the branch and version information from the given value,
+    which could be a single object or a list.
+    """
+    if lookup == 'in':
+        stripped_value = [_strip_object(el) for el in value]
+    else:
+        stripped_value = _strip_object(value)
+    return stripped_value
+
+
+# TODO: This code has been copies and pasted from openedx.core.djangoapps.xmodule_django
+# Need to figure out how can we access stuff from edx-platform.
```

Comment:
@Qubad786 We'll need to move the `CourseKeyField` functionality out of edx-platform into a common core repo.  @bmedx is making headway on this front by pulling out the `modulestore`, creating a new repo for the modulestore, and also creating a new edx-platform-core repo where some of the core/common functionality can go.

I imagine we'll want to move the entire `openedx/core/djangoapps/xmodule_django` app out of edx-platform so we can reuse this here.  Unless @clintonb or @rlucioni have already run into this in needing to store Course-Keys in SQL tables in other IDAs - and have created another library for this.

In the meantime, I'm glad you've added a `TODO` here to address this issue.  It would be good to include a JIRA ticket number with the TODO (and a back-pointer to this code) so we can keep track of this tech-debt.